web/pdf_viewer.js

```diff
@@ -2256,13 +2297,54 @@ class PDFViewer {
     if (!this.pdfDocument) {
       return;
     }
-    this.#annotationEditorMode = mode;
-    this.eventBus.dispatch("annotationeditormodechanged", {
-      source: this,
-      mode,
-    });
 
-    this.#annotationEditorUIManager.updateMode(mode, editId, isFromKeyboard);
+    const { eventBus } = this;
+    const updater = () => {
+      if (this.#onPageRenderedCallback) {
+        eventBus._off("pagerendered", this.#onPageRenderedCallback);
+        this.#onPageRenderedCallback = null;
+      }
```

Comment:
While the use of `signal` below removes the event listener itself when closing the current document, the `#onPageRenderedCallback`-field won't be removed so you probably want to `null` that one at the end of the `_resetView` method?